test/alphabet/nucleotide/nucl16_test.cpp

```diff
@@ -30,141 +30,233 @@
+// copy assignment
+TEST(nucl16, cp_assgn)
 {
-    constexpr nucl16 nucleotide1{nucl16::C};
-    constexpr nucl16 nucleotide2{std::move(nucleotide1)};
+    nucl16 t1{nucl16::C};
+    nucl16 t2;
+    nucl16 t3;
 
-    EXPECT_EQ(nucleotide2, nucl16::C);
+    t2 = t1;
+    t3 = t1;
+    EXPECT_EQ(t1, t2);
+    EXPECT_EQ(t2, t3);
 }
 
-TEST(alphabet_nucleotides_nucl16_test, copy_asn)
+// move assignment
+TEST(nucl16, mv_assgn)
 {
-    constexpr nucl16 nucleotide1{nucl16::D};
-    nucl16 nucleotide2{nucl16::H};
+    nucl16 t0{nucl16::C};
+    nucl16 t1{nucl16::C};
+    nucl16 t2;
+    nucl16 t3;
+    t2 = std::move(t1);
+    EXPECT_EQ(t2, t0);
+    t3 = std::move(t2);
+    EXPECT_EQ(t3, t0);
+}
 
-    nucleotide2 = nucleotide1;
+// swap
+TEST(nucl16, swap)
+{
+    nucl16 t0{nucl16::C};
+    nucl16 t1{nucl16::C};
+    nucl16 t2{};
+    nucl16 t3{};
 
-    EXPECT_EQ(nucleotide2, nucl16::D);
+    std::swap(t1, t2);
+    EXPECT_EQ(t2, t0);
+    EXPECT_EQ(t1, t3);
 }
 
-TEST(alphabet_nucleotides_nucl16_test, move_asn)
+// comparison
+TEST(nucl16, cmp)
 {
-    constexpr nucl16 nucleotide1{nucl16::K};
-    nucl16 nucleotide2{nucl16::G};
+    nucl16 t0{nucl16::A};
+    nucl16 t1{nucl16::C};
+    nucl16 t2{nucl16::G};
 
-    nucleotide2 = std::move(nucleotide1);
+    EXPECT_LT(t0, t1);
+    EXPECT_LE(t0, t1);
+    EXPECT_LE(t1, t1);
+    EXPECT_EQ(t1, t1);
+    EXPECT_GE(t1, t1);
+    EXPECT_GE(t2, t1);
+    EXPECT_GT(t2, t1);
+}
 
-    EXPECT_EQ(nucleotide2, nucl16::K);
+TEST(nucl16, to_char_member)
+{
+    for (size_t i = 0; i < 16; ++i)
+        EXPECT_EQ(all_nucl16[i].to_char(), capital_char[i]);
 }
 
-TEST(alphabet_nucleotides_nucl16_test, operator_char)
+TEST(nucl16, to_char_free)
 {
-    constexpr nucl16 nucleotide{nucl16::G};
-    constexpr char c = static_cast<char>(nucleotide);
+    for (size_t i = 0; i < 16; ++i)
+        EXPECT_EQ(to_char(all_nucl16[i]), capital_char[i]);
+}
 
-    EXPECT_EQ(c, 'G');
+TEST(nucl16, to_rank_member)
+{
+    for (size_t i = 0; i < 16; ++i)
+        EXPECT_EQ(all_nucl16[i].to_rank(), i);
 }
 
-TEST(alphabet_nucleotides_nucl16_test, to_char)
+TEST(nucl16, to_rank_free)
 {
-    uint8_t rank = 0;
-    for (auto i : capital_char)
-    {
-        nucl16 nucleotide{all_nucl16[rank++]};
-        char c = nucleotide.to_char();
-        EXPECT_EQ(c, i);
-    }
+    for (size_t i = 0; i < 16; ++i)
+        EXPECT_EQ(all_nucl16[i].to_rank(), i);
 }
 
-TEST(alphabet_nucleotides_nucl16_test, from_char)
+TEST(nucl16, stream_operator)
 {
-    uint8_t rank = 0;
-    for (auto i : capital_char)
-    {
-        nucl16 nucleotide{all_nucl16[rank++]};
-        EXPECT_EQ(nucl16{}.from_char(i), nucleotide);
-    }
+    std::stringstream ss;
+    ss << nucl16::A << nucl16::C << nucl16::G << nucl16::T << nucl16::U << nucl16::N << nucl16::UNKNOWN;
+    EXPECT_EQ(ss.str(), "ACGTUNN");
+}
 
-    rank = 0;
-    for (auto i : lower_char)
+TEST(nucl16, assign_char_member)
+{
+    nucl16 t0;
+    for (size_t i = 0; i < 16; ++i)
```

Comment:
I'm exhausted. Hopefully someone else can take the rest, or I will do the rest next week.